org.scala-ide.sdt.core/src/org/scalaide/core/internal/extensions/autoedits/CloseAngleBracketCreator.scala

```diff
@@ -0,0 +1,13 @@
+package org.scalaide.core.internal.extensions.autoedits
+
+import org.scalaide.core.text.Document
+import org.scalaide.core.text.TextChange
+import org.scalaide.extensions.autoedits.CloseAngleBracket
+
+object CloseAngleBracketCreator {
```

Comment:
I don't fully understand your comment. Generally, I agree that this code is ugly. The overall idea was to use the ToolBox and some runtime code generation to avoid these `*Creator` classes but so far I didn't find the time to implement it.